### PR TITLE
Added Snowflake provider to the Docker image

### PR DESCRIPTION
**Pull request by francescomucio**
The Snowflake database is quite popular and it makes sense to have its provider already available in the standard Docker image.

### CODE REVIEW

**Comment by potiuk** [comment]
I think static checks will fail and some fixes will need to be applied:

Specifically https://airflow.apache.org/docs/docker-stack/build-arg-ref.html should be updated

Also I think we need to add a Changelog entry in https://airflow.apache.org/docs/docker-stack/changelog.html to explain the change.

**Comment by francescomucio** [comment]
I did the changes in the doc files

File: docs/docker-stack/changelog.rst [hunk 1]
```diff
@@ -46,6 +46,9 @@ here so that users affected can find the reason for the changes.
 +--------------+---------------------+-----------------------------------------+------------------------+----------------------------------------------+
 | Date         | Affected images     | Potentially breaking change             | Reason                 | Link to Pull Request                         |
 +==============+=====================+=========================================+========================+==============================================+
+| 26 Jan 2023  | 2.5.1               | * Snoflake provider installed by        | Added the Snowflake    | https://github.com/apache/airflow/pull/29171 |
```
Comment (potiuk) [inline docs/docker-stack/changelog.rst]: It will be released in 2.6.0 i think as we are not likely to change the image with patchlevel.

Comment (potiuk) [reply docs/docker-stack/changelog.rst]: Docs and static checks stil fail :(

**Comment by potiuk** [comment]
Some failures to fix.

**Review by francescomucio — COMMENTED** [review]
Fixed typo

**Comment by potiuk** [comment]
> I am not sure how to fix static check that is requiring to regenerate the image with `breeze`.

The message printed is:

```
Image generation is needed. Please run this command:

breeze setup regenerate-command-images
```

Why not running the command ?

**Comment by francescomucio** [comment]
Where should I run it? I did the changes using the web UI in github.

Before writing this comment I have also cloned my fork on my machine and executed the command:

```bash
mucio@9i:~/airflow$ breeze setup regenerate-command-images

Reinstalling Breeze from /home/mucio/airflow/dev/breeze

Installing to existing venv 'apache-airflow-breeze'
  installed package apache-airflow-breeze 0.0.1, installed using Python 3.11.1
  These apps are now globally available
    - breeze
done! ✨ 🌟 ✨

Default value of python parameter 3.7 used.


Default value of backend parameter sqlite used.


Default value of postgres_version parameter 11 used.


Default value of mysql_version parameter 5.7 used.


Default value of mssql_version parameter 2017-latest used.

Skip generation of SVG images as command hashes are unchanged.
(allianz) mucio@9i:~/airflow$ git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
```

I have nothing else to commit. What am I missing?

**Comment by potiuk** [comment]
For any substantial changes, making a UI change is not nearly good enough. You need a development environment.

However if you want to use GitHub UI you can use codespaces (they are free up to 30 hours for everyone now and they open up the VS-code based editor that should likely be enough https://docs.github.com/en/codespaces/overview (and https://github.com/apache/airflow/blob/main/CONTRIBUTORS_QUICK_START_CODESPACES.rst linked from the general https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst where you see all other documentation and ways to setup your development environment - there are multiple ways - for example https://github.com/apache/airflow/blob/main/LOCAL_VIRTUALENV.rst explains all details about setting the local env)

Checking out your branch locally and setting up the environment there should be enough ( Are you sure you checked out your PR branch before you run the "regenerate" command ? You should also run `pre-commit install` (see details in contributing and then, when committing the code the static checks should fix itself.

**Comment by potiuk** [comment]
Also usually when you see a failure (like building docs or static checks) - the error message explain what you should do to replicate it locally (so just follow the advises)

**Comment by potiuk** [comment]
Do you still want to follow-up o that one @francescomucio ?

**Comment by francescomucio** [comment]
Yes, I will try to do it today or tomorrow

**Comment by eladkal** [comment]
can you please fix static checks errors?

**Comment by francescomucio** [comment]
@eladkal @potiuk finally found the time to fix it

**Comment by potiuk** [comment]
Cool. Cana you also please add a note in change log https://github.com/apache/airflow/blob/main/docs/docker-stack/changelog.rst#airflow-251 - it will be 2.6.0 version

**Review by potiuk — APPROVED** [review]

**Comment by francescomucio** [comment]
thanks a lot

**Pull request merged**